src/foam/lib/parse/Seq3.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.parse;
+
+public class Seq3
+  implements Parser
+{
+  private Parser[] parsers;
+  private int      index1;
+  private int      index2;
+  private int      index3;
+
+  public Seq3(int i, int j, int k,  Parser... args) {
```

Comment:
Why not make this generic and call it something like SeqI. Then take int[] indices. Then create an Object[] result of the same length. Then anyone can just specify any number of elements they want to pick out of the result. Or, just don't bother. Once I need to create an array for the result anyway, then it doesn't make too much difference if it is a few characters longer or not. It still needs to be allocated and garbage collected at that point, which is the majority of the cost.